### PR TITLE
Feature/df 197

### PR DESCRIPTION
**Pull request by AshTNA**
Hi @ababic 

Would you be able to merge this PR?

It fixes https://national-archives.atlassian.net/browse/DF-197 and uses constants and a loop to render out the buckets instead of duplicating the HTML to render each bucket.

Thanks :+1:

### CODE REVIEW

File: templates/search/blocks/search_landing_buckets.html [hunk 1]
```diff
@@ -2,39 +2,15 @@
...
-                </a>
-            </li>
+            {% for bucket_key, bucket_data in CATALOGUE_BUCKETS.items %}
+                <li class="search-landing-buckets__list-item" >
+                    {% bucket_count bucket_count_response bucket_data.count_key as bucket_count %}
+                    <a href="{% url 'search-catalogue' %}?group={{bucket_key}}" class="search-landing-buckets__link" >
+                        {{bucket_data.label}} ({{ bucket_count|intcomma }})
+                    </a>
+                </li>
+            {% endfor %}
```
Comment (ababic) [inline templates/search/blocks/search_landing_buckets.html:13]:
```suggestion
            {% for bucket in buckets %}
                <li class="search-landing-buckets__list-item" >
                    {% bucket_count bucket_count_response bucket_data.count_key as bucket_count %}
                    <a href="{% url 'search-catalogue' %}?group={{ bucket.key }}" class="search-landing-buckets__link" aria-current="{% if bucket == current_bucket %}true{% else %}false{% endif %}>
                        {{ bucket.label }} ({{ bucket_count|intcomma }})
                    </a>
                </li>
            {% endfor %}
```

File: templates/search/blocks/search_results.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
             {% with count=page.paginator.count per_page=page.paginator.per_page  %}
             <p class="search-results__metadata-count">Showing 
-                <span class="search-results__metadata-count-highlight">{% if count < per_page %}{{ count }}{% else %}{{ per_page }}{% endif %}</span> (of {{ count|intcomma }}) result{{ count|pluralize}} for "{{ form.q.value|default:'*' }}" in "{{ form.group.value }}"</p>
+                <span class="search-results__metadata-count-highlight">{% if count < per_page %}{{ count }}{% else %}{{ per_page }}{% endif %}</span> (of {{ count|intcomma }}) result{{ count|pluralize}} for "{{ form.q.value|default:'*' }}" in "{{ current_group_label }}"</p>
```
Comment (ababic) [inline templates/search/blocks/search_results.html:9]:
```suggestion
                <span class="search-results__metadata-count-highlight">{% if count < per_page %}{{ count }}{% else %}{{ per_page }}{% endif %}</span> (of {{ count|intcomma }}) result{{ count|pluralize}} for "{{ form.q.value|default:'*' }}" in "{{ current_bucket.label }}"</p>
```

File: templates/search/blocks/website_search_buckets.html [hunk 3]
```diff
@@ -3,47 +3,13 @@
...
-            </a>
-        </li>
+        {% for bucket_key, bucket_data in WEBSITE_BUCKETS.items %}
+            <li class="search-buckets__list-item--website" data-current="{% if form.group.value == bucket_key %}true{% else %}false{% endif %}">
+                {% bucket_count bucket_count_response bucket_data.count_key as bucket_count %}
+                <a href="?{% if form.q.value %}q={{ form.q.value }}&{% endif %}group={{bucket_key}}" class="search-buckets__link" aria-current="{% if form.group.value == bucket_key %}true{% else %}false{% endif %}">
+                    {{bucket_data.label}} ({{ bucket_count|intcomma }})
+                </a>
+            </li>
+        {% endfor %}
```
Comment (ababic) [inline templates/search/blocks/website_search_buckets.html:13]:
```suggestion
        {% for bucket in buckets %}
            <li class="search-buckets__list-item--website" data-current="{% if bucket == current_bucket %}true{% else %}false{% endif %}">
                {% bucket_count bucket_count_response bucket_data.count_key as bucket_count %}
                <a href="?{% if form.q.value %}q={{ form.q.value }}&{% endif %}group={{ bucket.key }}" class="search-buckets__link" aria-current="{% if bucket == current_bucket %}true{% else %}false{% endif %}">
                    {{ bucket.label }} ({{ bucket_count|intcomma }})
                </a>
            </li>
        {% endfor %}
```

**Review by ababic — CHANGES_REQUESTED** [review]
What you did works fine, but I think we could really benefit from introducing a new type here... I'm also very keen to get rid of that horrible `group:` prefix from all of the values (to match the updated filter fields).

**Comment by AshTNA** [comment]
Looks good, thanks @ababic

**Pull request merged**